src/Shipment.php

```diff
@@ -112,18 +114,37 @@ function (ResponseInterface $response) use ($request, $options) {
         };
     }
 
-    private function updateToken(RequestInterface $request)
+    private function getRefreshedToken(): string
     {
-        $token = $this->handleRefreshToken();
-        return \GuzzleHttp\Psr7\modify_request($request, [
-            'set_headers' => [
-                'Authorization' => 'Bearer ' . $token['access_token'],
-            ],
-        ]);
+        return $this->handleRefreshToken()['access_token'];
     }
 
     public function handleRefreshToken(): array
     {
         return $this->oAuth2->refreshToken($this->refreshToken);
     }
+
+    protected function createStack(): HandlerStack
+    {
+        return HandlerStack::create();
+    }
+
+    private function addRefreshTokenMiddlewareToStack(HandlerStack $stack): void
```

Comment:
Deixei uma sugestão de alterar o nome mas na real, acho que talvez nem seja necessário esse método. Vale o mesmo para o addRetryMiddlewareToStack. O motivo é o mesmo que citei para o método createStack.

Reply:
Prefiro deixar assim porquê facilita muito a leitura.

Acho que fica muito melhor que:
```
$this->stack = HandlerStack::create();

$this->stack->push(Middleware::retry($this->retryDecider(), $this->retryDelay()));

$this->stack->push($this->middlewareRefreshToken());
```

Lembrando que não quis fazer muitas refatorações aqui, porquê o objetivo era adicionar testes e corrigir alguns bugs descobertos, mas essa modificação é bem pequena e ajuda na sanidade mental de quem for ler